uniffi_bindgen/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+//! generated binding code. If you think of it as a kind of [wasm-bindgen](https://github.com/rustwasm/wasm-bindgen)
+//! wannabe, with a clunkier developer experience but support for more target languages,
+//! you'll be pretty close to the mark.
+//!
+//! Currently supported target languages include Kotlin, Swift and Python.
+//!
+//! ## Usage
+//
+//! To build a cross-language component using `uniffi`, follow these steps.
+//!
+//! ### 1) Specify your Component Interface
+//!
+//! Start by thinking about the interface you want to expose for use
+//! from other languages. Use the Interface Definition Language to specify your interface
+//! in a `.idl` file, where it can be processed by the tools from this crate.
+//! For example you might define an interface like this:
+//!
+//! ```text
+//! namespace example {
+//!   u32 foo(u32 bar);
+//! }
+//!
+//! dictionary MyData {
+//!   u32 num_foos;
+//!   bool has_a_bar;
+//! }
+//! ```
+//!
+//! ### 2) Implement the Component Interface as a Rust crate
+//!
+//! With the interface, defined, provide a corresponding implementation of that interface
+//! as a standard-looking Rust crate, using functions and structs and so-on. For example
+//! an implementation of the above Component Interface might look like this:
+//!
+//! ```text
+//! fn foo(bar: u32): u32 {
+//!     // TODO: a better example!
+//!     bar + 42
+//! }
+//!
+//! struct MyData {
+//!   num_foos: u32,
+//!   has_a_bar: bool
+//! }
+//! ```
+//!
+//! ### 3) Generate and include component scaffolding from the IDL file
+//!
+//! First you will need to install `uniffi-bindgen` on your system using `cargo install uniffi-bindgen`.
+//! Then add to your crate `uniffi_build` under `[build-dependencies]`.
+//! Finally, add a `build.rs` script to your crate and have it call [uniffi_build::generate_scaffolding](uniffi_build::generate_scaffolding)
+//! to process your `.idl` file. This will generate some Rust code to be included in the top-level source
+//! code of your crate. If your IDL file is named `example.idl`, then your build script would call:
+//!
+//! ```text
+//! uniffi_build::generate_scaffolding("./src/example.idl")
+//! ```
+//!
+//! This would output a rust file named `example.uniffi.rs`, ready to be
+//! included into the code of your rust crate like this:
+//!
+//! ```text
+//! include!(concat!(env!("OUT_DIR"), "/example.uniffi.rs"));
+//! ```
+//!
+//! ### 4) Generate foreign language bindings for the library
+//!
+//! The `uniffi-bindgen` utily provides a command-line tool that can produce code to
+//! consume the Rust library in any of several supported languages.
+//! It is done by calling (in python for example):
+//!
+//! ```text
+//! uniffi-bindgen --language python ./src/example.idl
+//! ```
+//!
+//! This will produce a file `example.py` in the same directory as the .idl file, containing python bindings
+//! to load and use the compiled rust code via its C-compatible FFI.
```

Comment:
If the python bindings are tier-two support level, we should probably not use python as the example language here.